src/js/components/BreadcrumbComponent.jsx

```diff
@@ -76,8 +165,16 @@ var BreadcrumbComponent = React.createClass({
   },
 
   render: function () {
+    var classSet = classNames({
+      breadcrumb: true,
+      collapsed: this.state.collapsed
+    });
+
+    // Hacky, but ensures that the dimension measurements are up-to-date
+    _.defer(this.updateDimensions);
```

Comment:
Yep, I hear what you're saying. `componentDidUpdate` is not the answer though, because it's generally called before the DOM has actually updated, and is therefore unhelpful for our purposes. The reason this `defer` call is in `render` rather than `componentDidUpdate` is purely pragmatic. 

I know the [real answer](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver), and I'd rather I didn't.